pytorch_lightning/trainer/supporters.py

```diff
@@ -457,6 +457,23 @@ def _wrap_loaders_max_size_cycle(self) -> Any:
             )
             state.reset()
 
+    def _apply_cycle_iterator_length(self) -> None:
+        """When the model is `max_size_cycle`, compute the length across all ``CycleIterator`` and re-assign it to
+        all dataloaders."""
+        if self.mode != "max_size_cycle":
+            return
+
+        all_lengths = apply_to_collection(
+            self.loaders, CycleIterator, lambda c: get_len(c.loader), wrong_dtype=(Sequence, Mapping)
```

Comment:
Wouldn't this work too?

```suggestion
            self.loaders, DataLoader, get_len
```

Also, why do you set `wrong_dtype`? Sequence and Mapping do not subclass CycleIterator

Reply:
Yes, removed !

Reply:
And it needs to apply on the CycleIterator which doesn't subclass DataLoader.